src/components/FlawAffects.vue

```diff
@@ -0,0 +1,1521 @@
+    isAffectSelected(affect));
+});
+
+const indeterminateSelection = computed(() => {
+  return !areAllAffectsSelected.value && paginatedAffects.value.filter(affect => isSelectable(affect)).some(affect =>
+    isAffectSelected(affect));
+});
+
+function isSelectable(affect: ZodAffectType) {
+  return !isBeingEdited(affect) && !isRemoved(affect);
+}
+
+function isAllNotSelectable() {
+  return paginatedAffects.value.every(affect => !isSelectable(affect));
+}
+
+function isAffectSelected(affect: any) {
+  return selectedAffects.value.includes(affect);
+}
+
+function toggleAffectSelection(affect: ZodAffectType) {
+  if (!isSelectable(affect)) {
+    return;
+  }
+  if (!isAffectSelected(affect)) {
+    selectedAffects.value.push(affect);
+  } else {
+    selectedAffects.value = selectedAffects.value.filter(a => a !== affect);
+  }
+}
+
+function selectAffects(event: Event) {
+  if (areAllAffectsSelected.value) {
+    paginatedAffects.value.forEach(affect => toggleAffectSelection(affect));
+  } else if (selectedAffects.value.length === 0) {
+    paginatedAffects.value.filter(affect => !isAffectSelected(affect)).forEach(affect => toggleAffectSelection(affect));
+  } else {
+    paginatedAffects.value.filter(affect => isAffectSelected(affect)).forEach(affect => toggleAffectSelection(affect));
+    (event.target as HTMLInputElement).checked = false;
+  }
+}
+
+// Affects Fields
+function affectCvss3Vector(affect: ZodAffectType) {
+  affect.cvss_scores.find(({ issuer, cvss_version }) => issuer === 'RH' && cvss_version === 'V3')
+    ?.vector
+    || null;
+}
+
+function resolutionOptions(affect: ZodAffectType) {
+  return affect?.affectedness
+    ? possibleAffectResolutions(affect?.impact)[affect.affectedness]
+    : [];
+}
+
+function affectRowTooltip(affect: ZodAffectType) {
+  if (isRemoved(affect)) {
+    return 'This affect will be deleted on save changes';
+  } else if (isNewAffect(affect)) {
+    return 'This affect will be created on save changes';
+  } else if (isModified(affect)) {
+    return 'This affect will be updated on save changes';
+  } else {
+    '';
+  }
+}
+
+// Affects Pagination
+const totalPages = computed(() =>
+  Math.ceil(sortedAffects.value.length / settings.value.affectsPerPage)
+);
+
+const minItemsPerPage = 5;
+const maxItemsPerPage = 20;
+const {
+  pages,
+  itemsPerPage,
+  currentPage,
+  changePage,
+} = usePagination(totalPages, settings.value.affectsPerPage);
+
+const paginatedAffects = computed(() => {
+  const start = (currentPage.value - 1) * settings.value.affectsPerPage;
+  const end = start + settings.value.affectsPerPage;
+  return sortedAffects.value.slice(start, end);
+});
+
+function reduceItemsPerPage() {
+  if (settings.value.affectsPerPage > minItemsPerPage) {
+    settings.value.affectsPerPage --;
+    itemsPerPage.value--;
+  }
+}
+
+function increaseItemsPerPage() {
+  if (settings.value.affectsPerPage < maxItemsPerPage) {
+    settings.value.affectsPerPage ++;
+    itemsPerPage.value++;
```

Comment:
If this two variables are in sync, could we remove `itemsPerPage` and only use `settings.value.affectsPerPage`?

Reply:
I didn't do that is cause it would imply to use `settings` value in `usePagination` on my initial implementation, which was making it less reusable and abstract. 

Rechecking this I notice that current implementation does not use `itemsPerPage` under `usePagination` and only use on components was the increment/decrement so it was removed and only the settings `perPage` value is now used